### PR TITLE
Linting and Optimizations

### PR DESCRIPTION
**Pull request by thanpolas**
@jpodwys would you entertain an API that supports promises?

### CODE REVIEW

File: cacheService.js [hunk 1]
```diff
@@ -232,17 +239,20 @@ function cacheService(cacheServiceConfig, cacheModules) {
   function checkCacheResponse(key, err, result, type, cacheIndex){
     if(err){
       log(true, 'Error when getting key ' + key + ' from cache of type ' + type + ':', err);
-      if(i < self.caches.length - 1){
-        return {status:'continue'};
-      }
+
+      // thanpolas: this comparison doens't make sense, "i" is not defined at this point.
+      //
+      // if(i < self.caches.length - 1){
```
Comment (thanpolas) [inline cacheService.js]: @jpodwys you need to have a look here, this is broken, what is the correct comparison you need to make?

**Comment by jpodwys** [comment]
Thanks for all the changes and issue reports. I'm not sure how much time I'll have today, but I'll look into these this week.

File: cacheService.js (hunk 1 above)
Comment (jpodwys) [inline cacheService.js]: Wow, great catch! This is likely supposed to compare `cacheIndex` rather than `i`.

**Comment by jpodwys** [comment]
I'm open to discussing supporting promises.

**Comment by thanpolas** [comment]
just pushed the fix for `cacheIndex`

**Comment by jpodwys** [comment]
All tests are passing. LGTM, thanks!

**Comment by jpodwys** [comment]
I'm thinking I'll wait to cut a new release until we've addressed some of the issues you reported this morning. However, if you need a new release cut ASAP, let me know.

**Pull request merged**